local-server/src/lib.rs

```diff
@@ -224,15 +227,44 @@ async fn linkup_request_handler(
 
     let extra_headers = get_additional_headers(&url, &headers, &session_name, &target_service);
 
-    if req
-        .headers()
-        .get("upgrade")
-        .map(|v| v == "websocket")
-        .unwrap_or(false)
-    {
-        handle_ws_req(req, target_service, extra_headers, client).await
-    } else {
-        handle_http_req(req, target_service, extra_headers, client).await
+    match ws.0 {
+        Some(upgrade) => {
+            let mut url = target_service.url;
```

Comment:
Maybe we can fix the naming here a bit to match the upstream/downstream terminology? I had a hard time spotting the ws_stream & upgrade distinction after not looking at the code for a while

Reply:
Do you mean the naming of `url` or the `upgrade`? 🤔

Reply:
I think that's right! "the next hop along" is upstream